src/popup/warnings/missing-permissions-alert.tsx

```diff
@@ -0,0 +1,85 @@
+import browser from 'webextension-polyfill'
+import { useAsyncFn } from 'react-use'
+import { Button } from '../components/button'
+import { useEffect } from 'react'
+import { Icon } from '../components/icon'
+import AlertIcon from '@fortawesome/fontawesome-free/svgs/solid/triangle-exclamation.svg'
+import { css } from '@emotion/react'
+import { palette } from '../theme'
+
+const PERMISSIONS = {
+  origins: ['*://*/*'],
+}
+
+const styles = {
+  root: css({
+    background: palette.yellow,
+    padding: '1rem',
+    marginBottom: '1rem',
+
+    '& > :first-child': {
+      marginTop: 0,
+    },
+    '& > :last-child': {
+      marginBottom: 0,
+    },
+  }),
+  icon: css({
+    margin: '0.25rem 0.5rem 0 0',
+    display: 'block',
+    float: 'left',
+  }),
+  button: css({
+    fontSize: '1rem',
+    display: 'block',
+    maxWidth: 'fit-content',
+    margin: 'auto',
+  }),
+}
+
+export function MissingPermissionsAlert(): JSX.Element | undefined {
+  const [{ value: hasPermission, loading }, checkPermission] = useAsyncFn(
+    async () => await browser.permissions.contains(PERMISSIONS),
+  )
+  useEffect(() => {
+    void checkPermission()
+  }, [checkPermission])
+
+  if (loading) {
+    return
+  }
+
+  if (hasPermission !== true) {
+    return (
+      <div css={styles.root}>
+        <p>
+          <Icon
+            css={styles.icon}
+            size='1.75rem'
+            title='alert'
+            theme='dark'
+            svg={AlertIcon}
+          />
```

Comment:
This icon has a hover effect. It's an issue with the `<Icon>` itself and its CSS. Basically, the hover was meant for buttons and links but was put on all icons. Technically, this is the first icon that's not a link so I guess it makes sense :shrug: